merge/conflict_test.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package merge_test
+
+import (
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/fieldpath"
+	"sigs.k8s.io/structured-merge-diff/merge"
+	"sigs.k8s.io/structured-merge-diff/value"
+)
+
+var (
+	// Short names for readable test cases.
+	_NS  = fieldpath.NewSet
+	_P   = fieldpath.MakePathOrDie
+	_KBF = fieldpath.KeyByFields
+	_SV  = value.StringValue
+	_IV  = value.IntValue
+)
+
+func TestNewFromSets(t *testing.T) {
+	got := merge.NewFromSets(map[string]*fieldpath.Set{
+		"Bob": _NS(
+			_P("key"),
+			_P("list", _KBF("key", _SV("a"), "id", _IV(2)), "id"),
+		),
+		"Alice": _NS(
+			_P("value"),
+			_P("list", _KBF("key", _SV("a"), "id", _IV(2)), "key"),
+		),
+	})
+	wanted := `conflicts with "Alice":
+- .value
+- .list[id=2,key="a"].key
```

Comment:
This test is a change detector if anything about the field path serialization changes :/

I guess it's not a huge problem, it just makes it hard to do things if there's a bunch of tests like this.

I'd probably just search the error message for the Alice & Bob lines if you want to make a followup, I don't feel super strongly.